subiquitycore/pubsub.py

```diff
@@ -33,3 +33,21 @@ async def abroadcast(self, channel):
 
     def broadcast(self, channel):
         return asyncio.get_event_loop().create_task(self.abroadcast(channel))
+
+
+class EventCallback:
```

Comment:
I'm still not completely happy about this as an abstraction but it's an improvement and will do for now :-)

It's not a problem for your usage here, but if one controller wants to subscribe to another controller's EventCallback it's not obvious how to do it without risking races today. Clearly you can create the EventCallback objects in the controller's __init__ methods but when do you subscribe? You can't do that in __init__ because other controller might not even exist yet, but if you do it in start() the other controllers start() method might already have been called and you might miss events. I guess there could be yet another method in the controller interface like "do_subscriptions" or something but that just makes things more complicated. Anyway, you've heard me go on like this enough already :)

Reply:
Sure thing.  Let's discuss next steps.  I put this on agenda for the week.